test/custom-error.js

```diff
@@ -0,0 +1,331 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/custom-error';
+
+const ruleTester = avaRuleTester(test, {
+	env: {
+		es6: true
+	},
+	parserOptions: {
+		sourceType: 'module'
+	}
+});
+
+const invalidClassNameError = {ruleId: 'custom-error', message: 'Invalid class name, use `FooError`.'};
+const constructorError = {ruleId: 'custom-error', message: 'Add a constructor to your error.'};
+const noSuperCallError = {ruleId: 'custom-error', message: 'Missing call to `super()` in constructor.'};
+const invalidNameError = {ruleId: 'custom-error', message: 'The `name` property should be set to `FooError`.'};
+const passMessageToSuperError = {ruleId: 'custom-error', message: 'Pass the error message to `super()`.'};
+const invalidMessageAssignmentError = {ruleId: 'custom-error', message: 'Pass the error message to `super()` instead of setting `this.message`.'};
+
+ruleTester.run('custom-error', rule, {
+	valid: [
+		'class Foo { }',
```

Comment:
Could you add:
- a test case with a class that extends something that is not an Error
- a test case with a class without name (to make sure it doesn't crash at any point), both when extending Error and not extending it (https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Classes#Class_expressions)
- a test case where the superclass is not an Identifier but something else: `class A extends Foo() {}` (just ignore it)
- a test case where the superclass is not an Identifier but a MemberExpression: `class A extends Foo.Bar {}`

For the last one, I think we could check if the property's name ends with Error (like `Http.ProtocolError`), but I'm fine ignoring it for know and add it if it comes up at some point. Anyway, this looks like it would crash the plugin at the moment.